src/extractor/guidance/turn_handler.cpp

```diff
@@ -459,124 +476,187 @@ Intersection TurnHandler::assignRightTurns(const EdgeID via_edge,
+        // fork candidates
         std::size_t left = best, right = best;
-        while (
-            left + 1 < intersection.size() &&
-            (angularDeviation(intersection[left + 1].angle, STRAIGHT_ANGLE) <= NARROW_TURN_ANGLE ||
-             (angularDeviation(intersection[left].angle, intersection[left + 1].angle) <=
-                  NARROW_TURN_ANGLE &&
-              angularDeviation(intersection[left].angle, STRAIGHT_ANGLE) <= GROUP_ANGLE)))
-            ++left;
-        while (
-            right > 1 &&
-            (angularDeviation(intersection[right - 1].angle, STRAIGHT_ANGLE) <= NARROW_TURN_ANGLE ||
-             (angularDeviation(intersection[right].angle, intersection[right - 1].angle) <
-                  NARROW_TURN_ANGLE &&
-              angularDeviation(intersection[right - 1].angle, STRAIGHT_ANGLE) <= GROUP_ANGLE)))
-            --right;
 
+        // find the leftmost road that might be part of a fork
+        while (left + 1 < number_of_connected_roads)
+        {
+            const auto angle_between_next_road_and_straight = angularDeviation(connected_roads[left + 1].angle, STRAIGHT_ANGLE);
+            const auto angle_between_next_road_and_left = angularDeviation(connected_roads[left].angle, connected_roads[left + 1].angle);
+            const auto angle_between_left_road_and_straight = angularDeviation(connected_roads[left].angle, STRAIGHT_ANGLE);
+
+            // check next left road if it is narrow to going straight
+            if (angle_between_next_road_and_straight <= NARROW_TURN_ANGLE)
+            {
+                ++left;
+            }
+            // check next left road if it is narrow to a road that can be seen as/grouped with going straight
+            else if (angle_between_next_road_and_left <= NARROW_TURN_ANGLE && angle_between_left_road_and_straight <= GROUP_ANGLE)
+            {
+                ++left;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        // find the rightmost road that might be part of a fork
+        while (right > 1)
+        {
+            const auto angle_between_right_road_and_straight = angularDeviation(connected_roads[right].angle, STRAIGHT_ANGLE);
+            const auto angle_between_prev_road_and_straight = angularDeviation(connected_roads[right - 1].angle, STRAIGHT_ANGLE);
+            const auto angle_between_prev_road_and_right = angularDeviation(connected_roads[right].angle, connected_roads[right - 1].angle);
+
+            if (angle_between_prev_road_and_straight <= NARROW_TURN_ANGLE)
+            {
+                --right;
+            }
+            else if (angle_between_prev_road_and_right <= NARROW_TURN_ANGLE && angle_between_right_road_and_straight <= GROUP_ANGLE)
+            {
+                --right;
+            }
+            else
+            {
+                break;
+            }
+        }
```

Comment:
I think you can express this and above more clearly using a adjacent-ish stdlib algorithm